web-portal/backend/prisma/schema.prisma

```diff
@@ -1,17 +1,135 @@
+  createdAt        DateTime
+  updatedAt        DateTime
+}
+
+model Tenant {
+  id        String   @id @unique @default(cuid())
+  active    Boolean  @default(true)
+  paymentLedger    PaymentLedger?
+  relayLedger RelayLedger?
+  keys Key[]
+  deleted   Boolean  @default(false)
+  createdAt DateTime
+  updatedAt DateTime
+}
+
 model User {
-  id    String  @id @default(cuid())
-  email String  @unique
-  name  String?
+  id        String   @id @unique @default(cuid())
+  org       Org      @relation(fields: [orgId], references: [id])
+  orgId     String
+  apps      App[]    @relation("UserApps")
+  active    Boolean  @default(true)
+  deleted   Boolean  @default(false)
+  createdAt DateTime
+  updatedAt DateTime
+}
+
+model App {
+  id        String    @id @unique @default(cuid())
+  org       Org       @relation(fields: [orgId], references: [id])
+  orgId     String
+  user      User      @relation("UserApps", fields: [userId], references: [id])
+  userId    String
+  appRules  AppRule[]
+  key String // api-key
+  active    Boolean   @default(true)
+  deleted   Boolean   @default(false)
+  createdAt DateTime
+  updatedAt DateTime
+}
+
+model AppRule {
+  id        String   @id @unique @default(cuid())
+  app       App      @relation(fields: [appId], references: [id])
+  appId     String
+  rule      RuleType?
+  value String
+  active    Boolean  @default(true)
+  deleted   Boolean  @default(false)
+  createdAt DateTime
+  updatedAt DateTime
+}
+
+model RuleType {
+  id              String   @id @unique @default(cuid())
+  name            String
+  isEditable      Boolean
+  isMultiple      Boolean
+  validationType  String
+  validationValue String
+  appRule   AppRule  @relation(fields: [appRuleId], references: [id])
+  appRuleId String          @unique
+  active          Boolean  @default(true)
+  deleted         Boolean  @default(false)
+  createdAt       DateTime
+  updatedAt       DateTime
+}
+
+model Key {
+  id        String   @id @unique @default(cuid())
+  appId     String
+  keyValue  String
+  tenant   Tenant  @relation(fields: [tenantId], references: [id])
+  tenantId String          @unique
+  active    Boolean  @default(true)
+  deleted   Boolean  @default(false)
+  createdAt DateTime
+  updatedAt DateTime
+}
+
+model PaymentLedger {
+  id               String          @id @unique @default(cuid())
+  tenant   Tenant  @relation(fields: [tenantId], references: [id])
+  tenantId String          @unique
+  referenceId      String
+  amount           Int
+  transactionType  TransactionType
+  createdAt        DateTime
+}
+
+enum TransactionType {
+  CREDIT
+  DEBIT
 }
+
+
+model RelayLedger {
+  id               String          @id @unique @default(cuid())
+  tenant   Tenant  @relation(fields: [tenantId], references: [id])
+  tenantId String          @unique
+  referenceId      String
+  amount           Int
+  chainId String
+  keyId String // fed from redis(redis-pg sync service // tbd
```

Comment:
Just for reference, this might be necessary to have as a reference to the id on the key table so we have consistancy (a string here that doesn't match a row in the key table would result in brokenness).